File: consumer/aisl-consumer-suite/packages/aisl-sdk-typescript/examples/object-context.mjs
import {AislClient} from '../dist/index.js';
const [apiUrl, systemId, objectId, revisionId] = process.argv.slice(2);
if (!apiUrl || !systemId || !objectId) throw new Error('usage: node examples/object-context.mjs <api-url> <system-id> <object-id> [revision-id]');
const client = new AislClient(apiUrl);
const revision = revisionId ? await client.revision(systemId, revisionId) : await client.activeRevision(systemId);
console.log(JSON.stringify(await revision.getDataModelObjectContext(objectId), null, 2));

File: consumer/aisl-consumer-suite/packages/aisl-sdk/examples/object_context.py
import argparse
import json
from aisl_sdk import AislClient

p = argparse.ArgumentParser()
p.add_argument("--api-url", required=True)
p.add_argument("--system-id", required=True)
p.add_argument("--revision-id")
p.add_argument("--object-id", required=True)
a = p.parse_args()

with AislClient(a.api_url) as client:
    revision = client.revision(a.system_id, a.revision_id) if a.revision_id else client.active_revision(a.system_id)
    print(json.dumps(revision.get_data_model_object_context(a.object_id), ensure_ascii=False, indent=2))
